src/robusta/integrations/discord/sender.py

```diff
@@ -0,0 +1,281 @@
+import requests
+from typing import Tuple
+from ...core.reporting.base import *
+from ...core.reporting.blocks import *
+from ...core.reporting.utils import add_pngs_for_all_svgs
+from ...core.model.env_vars import DISCORD_TABLE_COLUMNS_LIMIT, ROBUSTA_LOGO_URL
+
+ACTION_TRIGGER_PLAYBOOK = "trigger_playbook"
+DiscordBlock = Dict[str, Any]
+SEVERITY_EMOJI_MAP = {
+    FindingSeverity.HIGH: ":red_circle:",
+    FindingSeverity.MEDIUM: ":orange_circle:",
+    FindingSeverity.LOW: ":yellow_circle:",
+    FindingSeverity.INFO: ":green_circle:",
+}
+SEVERITY_COLOR_MAP = {
+    FindingSeverity.HIGH: "14495556",
+    FindingSeverity.MEDIUM: "16027661",
+    FindingSeverity.LOW: "16632664",
+    FindingSeverity.INFO: "7909721",
+}
+MAX_BLOCK_CHARS = 2000
+MAX_FIELD_CHARS = 1024
+BLANK_CHAR = "\u200b"
+
+class DiscordDescriptionBlock(BaseBlock):
+    """
+    Discord description block
+    """
+    description: str
+
+
+class DiscordFieldBlock(BaseBlock):
+    """
+    Discord field block
+    """
+    name: str
+    value: str
+    inline: bool
+
+    def __init__(self, name: str, value: str, inline: bool = False):
+        value = DiscordSender.apply_length_limit(value)
+        super().__init__(name=name, value=value, inline=inline)
+
+
+def _add_color_to_block(block: DiscordBlock, msg_color: str):
+    return {**block, "color": msg_color}
+
+
+class DiscordSender:
+    def __init__(
+            self, url: str, cluster_name: str
+    ):
+        """
+        Set the Discord webhook url.
+        """
+        self.url = url
+        self.cluster_name = cluster_name
+
+    @classmethod
+    def __add_log_level_icon(cls, title: str, severity: FindingSeverity) -> str:
+        icon = SEVERITY_EMOJI_MAP.get(severity, "")
+        return f"{icon} {severity.name} - {title}"
+
+    @staticmethod
+    def apply_length_limit(msg: str, max_length: int = MAX_FIELD_CHARS):
```

Comment:
We have an identical function in slack.
Can we extract it to a common location instead of having the same function twice?